bootstrap/pkg/kfapp/gcp/gcp.go

```diff
@@ -1675,14 +1709,28 @@ func (gcp *Gcp) setupWorkloadIdentity(namespace string) error {
 	return nil
 }
 
-// createK8sServiceAccount creates k8s servicea account with annotation
+// createOrUpdateK8sServiceAccount creates or updates k8s service account with annotation
 // iam.gke.io/gcp-service-account=gsa
 // TODO(lunkai): Ideally the k8s service account should be specified by kustomize.
-func createK8sServiceAccount(k8sClientset *clientset.Clientset, namespace string, name string, gsa string) error {
+func createOrUpdateK8sServiceAccount(k8sClientset *clientset.Clientset, namespace string, name string, gsa string) error {
 	log.Infof("Creating service account %v in namespace %v", name, namespace)
-	_, err := k8sClientset.CoreV1().ServiceAccounts(namespace).Get(name, metav1.GetOptions{})
+	currSA, err := k8sClientset.CoreV1().ServiceAccounts(namespace).Get(name, metav1.GetOptions{})
 	if err == nil {
 		log.Infof("Service account already exists...")
+		if currSA.Annotations == nil {
+			currSA.Annotations = map[string]string{
```

Comment:
nit: The code would be more robust as

if currSa.Annotation == nil {
   curSA.Annotations = map[string]string{}
}

currSa.Annotations[...] =....

This way you only end up writing this once.